contracts/validators/Allowlist.vy

```diff
@@ -0,0 +1,52 @@
+from ethereum.ercs import IERC20
+
+from .. import Validator
+implements: Validator
+
+MAX_PRODUCTS: constant(uint8) = 20
+
+owner: public(address)
+is_allowed: public(HashMap[address, bool])
+
+event Allowed:
+    user: indexed(address)
+
+event Denied:
+    user: indexed(address)
+
+
+@deploy
+def __init__(allowed: DynArray[address, 100]):
+    self.owner = msg.sender
+    for user: address in allowed:
+        self.is_allowed[user] = True
+        log Allowed(user)
```

Comment:
would it be cheaper to log once with an array of all users?

Reply:
oh yeah probably

Reply:
well... actually it's not as nice, if you are indexing on a particular user, since indexed arrays actually get hashed (so you can't index on a particular element in an array existing)